extension/persistence/eclipselink/src/main/java/org/apache/polaris/extension/persistence/impl/eclipselink/PolarisEclipseLinkStore.java

```diff
@@ -133,7 +133,7 @@ void deleteFromEntities(EntityManager session, long catalogId, long entityId) {
     diagnosticServices.check(session != null, "session_is_null");
     checkInitialized();
 
-    ModelEntity model = lookupEntity(session, catalogId, entityId);
+    ModelEntity model = lookupEntity(session, catalogId, entityId, -1);
```

Comment:
As far as I can tell all call paths to here come from `AbstractTransactionalPersistence.deleteEntityInCurrentTxn()` which has the entity's type code, so it could be propagated, I guess.

Reply:
Again, I'm not sure it makes sense on write paths. The idea here, as I interpret it, is to delete the entity with a given id, regardless of its type.

Reply:
These type codes were added from the WIP NoSQL implementation. In that code, entities are segregated by type in many cases. I do not think the NoSQL impl. does a "fan-out" search to delete entities having different type codes.

I tend to regard this type code parameter as having to reflect the correct actual type of the entity whose ID is passed as another parameter. Providing correct values is the responsibility of the caller.

If we want to change that aspect (and I'm open to discussing that), I'm not sure type codes will be helpful to the NoSQL impl.